test/unit/esm-loader.test.mjs

```diff
@@ -0,0 +1,181 @@
+/*
+ * Copyright 2022 New Relic Corporation. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import tap from 'tap'
+import sinon from 'sinon'
+import * as td from 'testdouble'
+
+tap.test('ES Module Loader', (t) => {
+  t.autoend()
+
+  let fakeSpecifier
+  let fakeContext
+  let fakeNextResolve
+  let fakeNewrelic
+  let fakeShimmer
+  let fakeLogger
+  let fakeLoggerChild
+  let loader
+
+  t.beforeEach(async () => {
+    fakeSpecifier = 'my-test-dep'
+    fakeContext = {}
+    fakeNextResolve = sinon.stub()
+
+    fakeLoggerChild = {
+      debug: sinon.stub()
+    }
+
+    fakeLogger = {
+      child: sinon.stub().returns(fakeLoggerChild)
+    }
+
+    fakeNewrelic = {
+      agent: {}
+    }
+
+    fakeShimmer = {
+      registerInstrumentation: sinon.stub(),
+      getInstrumentationNameFromModuleName: sinon.stub(),
+      registeredInstrumentations: {
+        express: {
+          moduleName: 'express',
+          type: 'web-framework',
+          onRequire: sinon.stub()
+        }
+      }
+    }
+
+    await td.replaceEsm('../../index.js', {}, fakeNewrelic)
+    await td.replaceEsm('../../lib/shimmer.js', {}, fakeShimmer)
+    await td.replaceEsm('../../lib/logger.js', {}, fakeLogger)
+
+    // eslint-disable-next-line node/no-unsupported-features/es-syntax
+    loader = await import('../../esm-loader.mjs')
```

Comment:
It's worth noting this is ignored because of https://github.com/mysticatea/eslint-plugin-node/issues/250.  I also logged a ticket to update our plugin so we can remove these: https://issues.newrelic.com/browse/NEWRELIC-3321